src/Pages/ADT3DScenePage/ADT3DScenePage.tsx

```diff
@@ -477,6 +531,43 @@ const ADT3DScenePageBase: React.FC<IADT3DScenePageProps> = ({
         }
     }, [setCorsPropertiesAdapterData?.adapterResult]);
 
+    // update the adx information of environment context with the fetched data
+    useEffect(() => {
+        if (!connectionState?.adapterResult.hasNoData()) {
+            const connectionData = connectionState.adapterResult.getData();
+
+            dispatch({
```

Comment:
same comment here, can we combine these actions into real "actions" that update multiple state variables in one shot? It seems reasonable that if you are setting the connection information, you would always want to set the status at the same time as well so maybe they can just be a single "StoreConnectionInformation" action.